crates/data_components/src/oracle/connection.rs

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2024-2025 The Spice.ai OSS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+     https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+use bb8_oracle::OracleConnectionManager;
+use oracle::Connector;
+use snafu::ResultExt;
+
+use crate::oracle::ConnectionSnafu;
+
+#[derive(Debug)]
+pub struct OracleConnectionPool {
+    pool: bb8::Pool<OracleConnectionManager>,
+}
+
+impl OracleConnectionPool {
+    pub async fn get(&self) -> super::Result<bb8::PooledConnection<'_, OracleConnectionManager>> {
+        let conn = self
+            .pool
+            .get()
+            .await
+            .map_err(|err| super::Error::ConnectionPoolError { source: err.into() })?;
+        Ok(conn)
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct OracleConnectionParams {
+    pub host: String,
+    pub port: u16,
+    pub username: String,
+    pub password: String,
+    pub service_name: String,
+}
```

Comment:
```suggestion
pub struct OracleConnectionParams {
    pub host: Arc<str>,
    pub port: u16,
    pub username: Arc<str>,
    pub password: Arc<str>,
    pub service_name: Arc<str>,
}
```

nit: if these don't change after they are built, can they be `Arc<str>`?

Reply:
Cool, I'll consider this approach, thank you!

Reply:
If it isn't beneficial, then can we remove the `Clone` derive? If we're planning to clone this config, then we should use Arc<str>

Reply:
Removed, thank you